### PR TITLE
PPTP-1844 | Align name field validation for full name and member name fields. 

### PR DESCRIPTION
**Pull request by ajchilts**
### Description of Work carried through

User Story
As a PPT user with a name that contains letters, hyphens, spaces, apostrophes and full stops
I want to be able to register on PPT

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [ ] Required Environment Config has been amended/added
 - [ ] Links to dependencies have been included (BE/FE work)
 - [x] User Acceptance Tests (UAT) were run locally and have passed.
 - [x] No Accessibility errors/warnings reported by Wave

### CODE REVIEW

File: test/uk/gov/hmrc/plasticpackagingtax/registration/views/group/ContactDetailsNameViewSpec.scala [hunk 1]
```diff
@@ -121,7 +162,48 @@ class ContactDetailsNameViewSpec extends UnitViewSpec with Matchers {
...
+        "Last name must only include letters, hyphens, spaces, apostrophes and full stops"
+      )
+    }
+
+    "last name is > 35" in {
+
+      val form = MemberName
+        .form()
+        .fillAndValidate(
+          MemberName("first", "hjghjgfhjdsgfhjdsgfdshjfgdsjhfgjshfgdshjfgdsjhfgdsjhfgdsjh")
```
Comment (liampan) [inline test/uk/gov/hmrc/plasticpackagingtax/registration/views/group/ContactDetailsNameViewSpec.scala:200]: this is difficult to count
```suggestion
          MemberName("first", "x" * 35)
```
or should that be 34

File: test/uk/gov/hmrc/plasticpackagingtax/registration/views/partner/PartnerMemberNamePageViewSpec.scala [hunk 2]
```diff
@@ -104,7 +105,48 @@ class PartnerMemberNamePageViewSpec extends UnitViewSpec with Matchers {
...
+        "First name must only include letters, hyphens, spaces, apostrophes and full stops"
+      )
+    }
+
+    "first name is > 35" in {
+
+      val form = MemberName
+        .form()
+        .fillAndValidate(
+          MemberName("hjghjgfhjdsgfhjdsgfdshjfgdsjhfgjshfgdshjfgdsjhfgdsjhfgdsjh", "last")
```
Comment (liampan) [inline test/uk/gov/hmrc/plasticpackagingtax/registration/views/partner/PartnerMemberNamePageViewSpec.scala:143]: ^^

**Review by liampan — APPROVED** [review]
Looks good a few lil bits of refactor

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/CommonFormValidators.scala [hunk 3]
```diff
@@ -24,14 +24,24 @@ import java.util.regex.Pattern
...
-      .verifying(errorKey, _.nonEmpty)
-      .transform[String](_.get, Some.apply)
-      .verifying(errorKey, _.trim.nonEmpty)
+  val fullNameRegexPattern: Pattern     = Pattern.compile("^[A-Za-z0-9 ,.()/&''-]{1,160}$")
+  val nameRegexPattern: Pattern         = Pattern.compile("^[A-Za-z0-9 ,.()/&''-]{1,35}$")
+  val phoneNumberRegexPattern: Pattern  = Pattern.compile("^[+]?[0-9 ]*$")
+  val addressInputRegexPattern: Pattern = Pattern.compile("^[A-Za-z0-9 \\-,.&']{1,35}$")
+
+  val emailPattern: Pattern =
+    Pattern.compile("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\\.[a-zA-Z0-9-]+)*$")
```
[the remark this answers is not in the document]
Comment (ajchilts) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/forms/CommonFormValidators.scala:33]: First time I have seen this. I would recommend that it be discussed and if required; that it be done on a separate ticket? This ticket was for names only. The only change to the above regex was to move it to the top of the file.

[the remark this answers is not in the document]
Comment (liampan) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/forms/CommonFormValidators.scala:33]: Yep deffo not a blacker for this ticket

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/CommonFormValidators.scala [hunk 4]
```diff
@@ -24,14 +24,24 @@ import java.util.regex.Pattern
 
 trait CommonFormValidators {
 
-  def nonEmptyString(errorKey: String): Mapping[String] =
-    optional(text)
-      .verifying(errorKey, _.nonEmpty)
-      .transform[String](_.get, Some.apply)
-      .verifying(errorKey, _.trim.nonEmpty)
+  val fullNameRegexPattern: Pattern     = Pattern.compile("^[A-Za-z0-9 ,.()/&''-]{1,160}$")
+  val nameRegexPattern: Pattern         = Pattern.compile("^[A-Za-z0-9 ,.()/&''-]{1,35}$")
```
[the remark this answers is not in the document]
Comment (ajchilts) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/forms/CommonFormValidators.scala:28]: No @PSouthern  as they are not allowed in the EIS schema for first name and last name. So if we let them through for first name and last name we would get a 400 error on submission. 

In the EIS schema full name uses the company name regex which does allow accented chars however; we have aligned full name with first name and last name fields so we do not have mixed behaviour around names. Given the restrictions in the EIS schema we can't allow accented names for both. therefore the sensible approach is to align full name with first name and last name validation and give it the 160 length to allow for the combined names.

[the remark this answers is not in the document]
Comment (PSouthern) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/forms/CommonFormValidators.scala:28]: Makes sense, thanks for the clarification 👍

**Pull request merged**